### PR TITLE
Do not emit a type test for exception object to be an object when the…

### LINKED ISSUE #4620: Catch filters without a formal catch type do not need to test exception to be Object
ECMA335 specifies that everything that can be caught derives from Object.

I.12.4.2 Exception handling:
. . . All exception objects are instances of some class (i.e., they can be boxed value types, but not pointers, unboxed value types, etc.). . . .

So, it is either we are targeting a conforming CLI implementation and object test is redundant, or we are targeting a system where exceptions are not necessarily objects and object check is still not a good idea.

### PR DESCRIPTION
**Pull request by VSadov**
…re is no formal catch type.

Fixes #4620

### CODE REVIEW

**Comment by VSadov** [comment]
test this please

**Comment by VSadov** [comment]
@gafter @AlekseyTs @jaredpar @agocke - please take a look. Trivial fix.

**Comment by AlekseyTs** [comment]
LGTM, assuming VS2013 didn't do the check too.

**Comment by VSadov** [comment]
VS2013 did not support filters in C#. This is a C# only change.

**Comment by VSadov** [comment]
CC @tmat

File: src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs [hunk 1]
```diff
@@ -866,13 +867,22 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
...
+
+                    _builder.EmitOpCode(ILOpCode.Isinst);
+                    _builder.EmitToken(exceptionType, catchBlock.Syntax, _diagnostics);
+                    _builder.EmitOpCode(ILOpCode.Dup);
+                    _builder.EmitBranch(ILOpCode.Brtrue, typeCheckPassedLabel);
+                    _builder.EmitOpCode(ILOpCode.Pop);
+                    _builder.EmitIntConstant(0);
+                    _builder.EmitBranch(ILOpCode.Br, typeCheckFailedLabel);
+                }
+                else
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs]: Can we delete the empty else block

**Pull request merged**